docs/tutorials/semantic_search/asymmetric_embedding_model.md

```diff
@@ -0,0 +1,539 @@
+
+Run the following command in the directory containing these files:
+
+```
+zip -r intfloat-multilingual-e5-small-onnx.zip model.onnx tokenizer.json sentencepiece.bpe.model
+```
+
+This command will create a zip file named `intfloat-multilingual-e5-small-onnx.zip`, with the previous mentioned files.
+
+### c. Calculate the Model File Hash
+
+Before registering the model, you need to calculate the SHA-256 hash of the zip file. Run this command to generate the hash:
+
+```
+shasum -a 256 intfloat-multilingual-e5-small-onnx.zip
+```
+
+Make a note of the hash value, as you will need it during the model registration process.
+
+### d. Serve the Model File Using a Python HTTP Server
+
+To allow OpenSearch to access the model file, you need to serve it via HTTP. Since this is a local development environment, you can use Python's built-in HTTP server:
+
+Navigate to the directory containing the zip file and run the following command:
+
+```
+python3 -m http.server 8080 --bind 0.0.0.0
+```
+
+This will serve the zip file at `http://0.0.0.0:8080/intfloat-multilingual-e5-small-onnx.zip`. After registering the model, you can stop the server by pressing `Ctrl + C`.
+
+---
+
+## Step 3: Register a Model Group
+
+Before registering the model itself, you need to create a model group. This helps organize models in OpenSearch. Run the following request to create a new model group:
+
+```
+POST /_plugins/_ml/model_groups/_register
+{
+  "name": "Asymmetric Model Group",
+  "description": "A model group for local asymmetric models"
+}
+```
+
+Take note of the `model_group_id` returned in the response, as it will be required when registering the model.
+
+---
+
+## Step 4: Register the Model
+
+Now that you have the model zip file and the model group ID, you can register the model in OpenSearch. Run the following request:
+
+```
+POST /_plugins/_ml/models/_register
+{
+    "name": "e5-small-onnx",
+    "version": "1.0.0",
+    "description": "Asymmetric multilingual-e5-small model",
+    "model_format": "ONNX",
+    "model_group_id": "your_group_id",
+    "model_content_hash_value": "your_model_zip_content_hash_value",
+    "model_config": {
+        "model_type": "bert",
+        "embedding_dimension": 384,
+        "framework_type": "sentence_transformers",
+        "query_prefix": "query: ",
+        "passage_prefix": "passage: ",
+        "all_config": "{ \"_name_or_path\": \"intfloat/multilingual-e5-small\", \"architectures\": [ \"BertModel\" ], \"attention_probs_dropout_prob\": 0.1, \"hidden_size\": 384, \"num_attention_heads\": 12, \"num_hidden_layers\": 12, \"tokenizer_class\": \"XLMRobertaTokenizer\" }"
+    },
+    "url": "http://host.docker.internal:8080/intfloat-multilingual-e5-small-onnx.zip"
+}
+```
+
+Replace `your_group_id` and `your_model_zip_content_hash_value` with the actual values from earlier. This will initiate the model registration process, and you’ll receive a task ID in the response.
```

Comment:
```suggestion
Replace `your_group_id` and `your_model_zip_content_hash_value` with the values from previos steps. This will initiate the model registration process, and you’ll receive a task ID in the response.
```